tests/Operation/ExplainTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Operation\Count;
+use MongoDB\Operation\Distinct;
+use MongoDB\Operation\Explain;
+
+class ExplainTest extends TestCase
+{
+    /**
+     * @expectedException MongoDB\Exception\InvalidArgumentException
+     * @dataProvider provideInvalidConstructorOptions
+     */
+    public function testConstructorOptionTypeChecksForCount(array $options)
+    {
+        $explainable = new Count($this->getDatabaseName(), $this->getCollectionName(),[]);
```

Comment:
Since you know that constructor option validation doesn't care about the Explainable's concrete type, you shouldn't need separate tests for each possibility. If the constructor _did_ care for some reason, we'd certainly want to test that.

We can use PHPUnit to create a mock object from the interface itself:

```php
$explainable = $this->createMock('MongoDB\Operation\Explainable');
```

For more info on this, see [Test Doubles](https://phpunit.de/manual/current/en/test-doubles.html) in the PHPUnit docs.